File: abc017_b.kt

fun main( args : Array<String> ) { 
  readLine()!!.let { 
    val a = it.replace("o", "X")
              .replace("k", "X")
              .replace("u", "X")
              .replace("ch", "X")
              .replace("X", "")
    if ( a == "" )
      println("YES")
    else 
      println("NO")

  }
}

File: abc017_b.py

import collections
S=input()
que = collections.deque(S)

while que:
    lc = que.pop()
    if lc in {"o", "k", "u"}:
        continue
    elif lc == "h" and len(que) > 0 and que[-1] == "c":
        que.pop()
        continue
    else:
        print("NO")
        exit()
print("YES")
